### DISCUSSION TITLE
[Solved] Inbuilt themes are not working and installed themes are not showing up.

### DISCUSSION THREAD

**Discussion by dir1337** [lxqt/lxqt#2476] [Q&A] [answered]
**Issue:**
In the newly installed LXQT desktop environment. From preferences>LXQT settings> Appearance, it is unable to change themes, as shown in the website as examples  the (inbuilt) wallpapers, icons etc. It's appearance is  with some icons and no wallpaper. 
And the installed adapta-project themes also not shown as the option.

**Need suggestion**
Some changes occurs only restarting the desktop. Is the any handy command for reload or something. 

**Question**
What changes I should do to make it working?

**Comment by stefonarch** [comment]
You should provide more details, distribution, what exactly did you install, other DEs installed.

**Reply by dir1337** [reply]
I found while trying lxappearance : the command not found.
So, I did to find this out.
> `ls /usr/bin/ | grep -iE "lx"`
lxdoas
lximage-qt
lxqt-about
lxqt-admin-time
lxqt-admin-user
lxqt-admin-user-helper
lxqt-archiver
lxqt-backlight_backend
lxqt-config
lxqt-config-appearance
lxqt-config-brightness
lxqt-config-file-associations
lxqt-config-globalkeyshortcuts
lxqt-config-input
lxqt-config-locale
lxqt-config-monitor
lxqt-config-notificationd
lxqt-config-powermanagement
lxqt-config-session
lxqt-connman-applet
lxqt-globalkeysd
lxqt-leave
lxqt-notificationd
lxqt-openssh-askpass
lxqt-panel
lxqt-policykit-agent
lxqt-powermanagement
lxqt-runner
lxqt-session
lxqt-sudo
lxsu
lxsudo
startlxqt

**Comment by dir1337** [comment]
Distribution : Arch 
command : sudo  pacman -S lxqt
session starts with 'startx'
![icons_show_error](https://github.com/lxqt/lxqt/assets/144369627/fae20188-174a-4329-93d3-9412ed307e76)
![appearance_applet_error](https://github.com/lxqt/lxqt/assets/144369627/e31357e0-13a3-4664-b615-89050b9ba2c1)


**Extra installed packages :**
LXQt Connman applet , nm-tray , libstatgrab, libsysstat.  


**Themes dir**
>ls /usr/share/themes/
`Adapta, Bear2, Emacs, Onyx-Citrus
Adapta-Eta, Clearlooks, Mikachu, Orang
Adapta-Nokto, Clearlooks-3.4, Natura, Raleigh
Adapta-Nokto-Eta, Clearlooks-Olive, Nordic, Syscrash
Artwiz-boxed, Default, Onyx`

All the above mention themes are available for openbox-3 but not for desktop. 

That's it.
Any guidance how to configure. (the icons are not showing and the installed theme is not shown up in the applet).

**Reply by tsujan** [reply]
You have just one icon theme, which is very poor BTW. If you want that dialog to show more icon themes, you should first install them (others might tell you their package names in Arch).

Also, if you've installed `lxqt-themes`, you should see LXQt themes in the above dialog. They are for the panel and runner.

As for the widget style (the look and feel of app windows), you use Kvantum, which comes with many themes. They can be seen and changed in Kvantum Manager. If you want another widget style, change it with the same dialog -- provided that you've installed another style (Breeze, QtCurves,...).

All in all, apparently your problem is that you haven't installed any icon set.

**Reply by dir1337** [reply]
It's late. I thought the problem lies in repo package (some version mismatch) so i remove it. And newly/freshly install LXQT from source. So now the problem has increased. The panel is also gone from bottom and only openbox menu-bar is working. While applications are working form the terminal.

**dmesg:**
https://0x0.st/HfGj.xyz


**Found**

Also every time entering in, the prompt of "desktop crashes and autostart fails", "runner crashes and autostart fails". "power-manager crashes and autostart fails " welcomes me there.
![new_error_after_newly_installation](https://github.com/lxqt/lxqt/assets/144369627/25aa040b-c861-4b0d-9a45-95e973d88b32)



A black blank screen but keybinding are working.
Any suggestion what i am missing from my side, to configure it.

**Reply by stefonarch** [reply]
Maybe you are running an openbox session. If you don't use a display manager like sddm you need in `~/.xinitrc`

`exec startlxqt`

**Reply by dir1337** [reply]
Yes your are correct, I am not using display manager( is it necessary with it functionally work? I guess not.) I am starting it with .xinitrc file.

**Environment Variable:**
LD_LIBRARY_PATH=/usr/local/lib
Result me with panel. So panel is set.
But 
**Error:**
qterminal segfault (core dump).

So is there more to configure then just to install the package from source. Please let me know if there are any need to set environment variables any references.
Because as suggested lxqt-themes is already installed so where is lxqt searching for it do I need to explicitly mention the path by setting envs.
All the wallpaper,themes,icons are here:
/usr/local/share/lxqt
But why they are  be ignored.

**Reply by stefonarch** [reply]
If I understood you right you installed it with the build script, which accepts variables to install in /usr/ instead of /usr/local. LXQt Themes should be in` /usr/local/share/lxqt/ in your installation.
`
But I've no idea what exactly happened with your setup, it looks really overcomplicated. On arch installing `lxqt` and select` LXQt Desktop` in the display manager is all which is needed . Using sddm or startx+xinitrc should be the same.

**Reply by dir1337** [reply]
(qterminal prob is solved ).
Only the desktop crashes warning says that lxqt-config > desktop  is not working.
Even mentioning the path of  a specific image it ignores it, that mean it is not working, else every thing is working. 
So the problem lies in the desktop. 

![lxqt_desktop_error](https://github.com/lxqt/lxqt/assets/144369627/22ecebdf-d1e2-4603-a3e3-89f82827b090)

**Reply by isf63** [reply]
What distribution are you using? Could be that the distribution's packagers haven't carefully integrated LXQt.

**Reply by tsujan** [reply]
> So the problem lies in the desktop.

Most probably, it's in your build.

**Reply by dir1337** [reply]
@isf63, Most of the time my first choice is that one, just pacman/yay (to save time). https://github.com/lxqt/lxqt/discussions/2476#discussioncomment-6981164

**Reply by dir1337** [reply]
@tsujan Thank you, at least you are trying hard to figure it out (as Maintainer).

**Reply by dir1337** [reply]
**backtrace:** 
Core was generated by `/usr/local/bin/pcmanfm-qt --desktop --profile=lxqt'.
Program terminated with signal SIGSEGV, Segmentation fault.
0.  0x000055c9a4ffcbc5 in PCManFM::DesktopWindow::updateFromSettings (
    this=0x55c9a7066c70, settings=..., changeSlide=true)
    at /home/xyz/installed_packages/lxqt/pcmanfm-qt/pcmanfm/desktopwindow.cpp:898
898 selectionModel()->clearCurrentIndex();
[Current thread is 1 (Thread 0x7f6975e51880 (LWP 804))]
**(gdb) bt**
0.  0x000055c9a4ffcbc5 in PCManFM::DesktopWindow::updateFromSettings(PCManFM::Settings&, bool) (this=0x55c9a7066c70, settings=..., changeSlide=true)
    at /home/xyz/installed_packages/lxqt/pcmanfm-qt/pcmanfm/desktopwindow.cpp:898
1.  0x000055c9a4f9f2eb in PCManFM::Application::createDesktopWindow(int)
    (this=0x7ffe9c4b21a0, screenNum=1)
    at /home/xyz/installed_packages/lxqt/pcmanfm-qt/pcmanfm/application.cpp:820
2.  0x000055c9a4f9cf6c in PCManFM::Application::desktopManager(bool)
    (this=0x7ffe9c4b21a0, enabled=true)
    at /home/xyz/installed_packages/lxqt/pcmanfm-qt/pcmanfm/application.cpp:471
3.  0x000055c9a4f9b591 in PCManFM::Application::parseCommandLineArgs()
    (this=0x7ffe9c4b21a0)
    at /home/xyz/installed_packages/lxqt/pcmanfm-qt/pcmanfm/application.cpp:247
4.  0x000055c9a4f9c490 in PCManFM::Application::exec() (this=0x7ffe9c4b21a0)
    at /home/xyz/installed_packages/lxqt/pcmanfm-qt/pcmanfm/application.cpp:346
5.  0x000055c9a4faa64d in main(int, char**) (argc=3, argv=0x7ffe9c4b2548)
    at /home/xyz/installed_packages/lxqt/pcmanfm-qt/pcmanfm/pcmanfm.cpp:13

**Reply by tsujan** [reply]
Make sure you haven't mixed git sources with released ones. The sources of `pcmanfm-qt` and `libfm-qt` should match.

**Reply by tsujan** [reply]
Also, make sure you don't have double installations of anything, one in `/usr/local/` and the other in `/usr/`.

**Reply by dir1337** [reply]
**Form journal** 
https://0x0.st/HOb2.xyz
> Make sure you haven't mixed git sources with released ones. The sources of `pcmanfm-qt` and `libfm-qt` should match.

How ?

**Reply by tsujan** [reply]
> Form journal

The crash isn't reproducible. My guess is that something has gone wrong during compilation.

I wonder why you don't use the existing Arch packages, like others do; they should have no problem. So much trouble for what?! You don't need git; do you?

> How ?

If you made installable packages first, there's no need to that — `pacman` has taken care of it. But if you installed directly from the source (not recommended), by checking both directories.

**Reply by dir1337** [reply]
>I wonder why you don't use the existing Arch packages, like others do; they should have no problem. So much trouble for what?! You don't need git; do you?

https://github.com/lxqt/lxqt/discussions/2476#discussioncomment-6981164 .I already mentioned that I "First Use" pacman.
Same problem occurred there also.

**Reply by tsujan** [reply]
Instead of letting people help you with the standard Arch packages, you jumped into self-compiling and created more problems for yourself. Now no one can know what problem is caused by what.

**Reply by dir1337** [reply, marked as answer]
I don't know why u are not trying to understand that "First I Used Arch Package" (simple), the same problem lies there also. There is nothing related to my build. If you think building your package is that much difficult on the earth, then please remove the build instructions from Readme files, provide binaries instead.
But understand, it is not okay.

Anyways 
Thank you for your support @tsujan @stefonarch .

**Solution:**
The problem was with the graphic card configuration with xorg conf file ( which was set to default but that's not how it works ) deleting file xorg.conf and files from xorg.conf.d directory works fine ( for the new arch machine ).
This issue was solved here: https://bbs.archlinux.org/viewtopic.php?id=288866

**Reply by stefonarch** [reply]
Happy you got this help - as we always thought it had nothing to do with LXQt itself.

**Reply by tsujan** [reply]
> There is nothing related to my build...

Those who might offer help can't be sure of that; instead, they know that Arch packages are OK.

Compilation doesn't solve anything, unless suggested by devs. This is yet another example of that.
